src/core/Rational.pm6

```diff
@@ -94,35 +94,38 @@ my role Rational[::NuT = Int, ::DeT = ::("NuT")] does Real {
         my $whole  = self.abs.floor;
         my $fract  = self.abs - $whole;
 
+        my $reduce = False;
         # fight floating point noise issues RT#126016
-        if $fract.Num == 1e0 { ++$whole; $fract = 0 }
+        if $fract.Num == 1e0 {
+            if nqp::unbox_s(self.^name) eq 'Rat' {
```

Comment:
I think this could/should be `if nqp::eqaddr(self.WHAT,Rat) {`. Then you're actually checking against the type itself, not the string representation of its name.

Reply:
I am not arguing, but I did just copy that syntax from line 14 in the same file. I can certanly change that if it is a better way to go.

Reply:
Yeah, but that line is intentionally creating a string. Line 12 does `nqp::eqaddr(self.WHAT,Rational),` because there it's checking for a type. If you grep through the Rakudo sources you'll see a whole lot more uses of `nqp::eqaddr(<something>.WHAT, <some type>)` than `<something>.^name eq <some type name>`. What you have would work, it's just slower.